server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -635,6 +638,8 @@ public Iterator<Setting<?>> settings() {
     private final Double writeLoadForecast;
     @Nullable
     private final Long shardSizeInBytesForecast;
+    // Key: model ID, Value: Fields that use model
+    private final Map<String, Set<String>> fieldsForModels;
```

Comment:
I changed the name of the variable to more clearly indicate that the model ID is the key and the field set is the value

Reply:
sounds good. I went so many times backward and forward on this one that I probably was not consistent / clear enough on the PoC 👍